frontend/rust-lib/flowy-grid/src/event_map.rs

```diff
@@ -42,63 +42,136 @@ pub fn create(grid_manager: Arc<GridManager>) -> Module {
+    /// for more information.
+    ///
+    /// The event handler accepts [UpdateFieldTypeOptionPayloadPB] and return errors if failed to modify the
+    /// field.
     #[event(input = "UpdateFieldTypeOptionPayloadPB")]
     UpdateFieldTypeOption = 12,
 
+    /// [InsertField] event is used to insert a new field. If the field is already exists, the event
+    /// handler will replace the value with the new field value.
     #[event(input = "InsertFieldPayloadPB")]
     InsertField = 13,
 
+    /// [DeleteField] event is used to delete a new field. [DeleteFieldPayloadPB] is the context that
+    /// is used to delete the field from the Grid.
     #[event(input = "DeleteFieldPayloadPB")]
     DeleteField = 14,
 
+    /// [SwitchToField] event is used to update the current field's type.
+    /// It will insert a new FieldTypeOptionData if the new FieldType doesn't exist before, otherwise
+    /// reuse the existing FieldTypeOptionData. You could check the [GridRevisionPad] for more details.
     #[event(input = "EditFieldPayloadPB", output = "FieldTypeOptionDataPB")]
     SwitchToField = 20,
 
+    /// [DuplicateField] event is used to duplicate the field. The duplicated field data is kind of
+    /// deep copy of the target field. The passed in [DuplicateFieldPayloadPB] is the context that is
+    /// used to duplicate the field.
+    ///
+    /// Return errors if failed to duplicate the field.
+    ///
     #[event(input = "DuplicateFieldPayloadPB")]
     DuplicateField = 21,
 
+    /// [MoveItem] event is used to move the item. For the moment, the item has two types defined in
+    /// the [MoveItemTypePB].
     #[event(input = "MoveItemPayloadPB")]
     MoveItem = 22,
 
+    /// [GetFieldTypeOption] event is used to get the FieldTypeOption data for the specific field type.
+    ///
+    /// Check out the [FieldTypeOptionDataPB] for more details. If the [FieldTypeOptionData] does exist
+    /// for the target type, the [TypeOptionBuilder] will create the default data for that type.
+    ///
+    /// Return the [FieldTypeOptionDataPB] if there is no errors.
     #[event(input = "GridFieldTypeOptionIdPB", output = "FieldTypeOptionDataPB")]
     GetFieldTypeOption = 23,
 
+    /// [CreateFieldTypeOption] event is used to create a new FieldTypeOptionData.
     #[event(input = "CreateFieldPayloadPB", output = "FieldTypeOptionDataPB")]
     CreateFieldTypeOption = 24,
 
+    /// [NewSelectOption] event is used to create a new select option. Return the [SelectOptionPB] if
+    /// there is no errors.
     #[event(input = "CreateSelectOptionPayloadPB", output = "SelectOptionPB")]
     NewSelectOption = 30,
 
+    /// [GetSelectOptionCellData] event is used to get the select option data for cell editing.
+    /// [GridCellIdPB] locate which cell data that will be read from. The return value, [SelectOptionCellDataPB]
+    /// contains the available options and the currently selected options.
```

Comment:
/// [GetSelectOptionCellData] event is used to get the select option data for cell editing.
/// [GridCellIdPB] defines from which cell the data will be read. The return value, [SelectOptionCellDataPB]
/// contains the available options and the currently selected options.